exercise/cache/exercise.py

```diff
@@ -14,6 +14,8 @@ class ExerciseCache(CachedAbstract):
     def __init__(self, exercise, language, request, students, url_name):
         self.exercise = exercise
         self.load_args = [language, request, students, url_name]
+        self.compressor = lzma.compress
```

Comment:
Because these variables shouldn't be defined for specific exercise, I would move them to the module globals or I would set them as class attributes. For the former, you could do `from lzma import compress as compressor, decompress as decompressor`. For the latter:

```
class ExerciseCache(CachedAbstract):
  compressor = lzma.compress
  decompressor = lzma.decompress

  def content(self):
    return self.decompressor(self.data['content']).decode('utf-8')
```